src/adapter/src/coord.rs

```diff
@@ -2191,27 +2186,38 @@ impl Coordinator {
         // For compute reconciliation to recognize that an existing dataflow can be reused, we want
         // to advance the `as_of` far enough that it is beyond the `as_of`s of all dataflows that
         // might still be installed on replicas, but ideally not much farther as that would just
-        // increase the wait time until the index becomes readable. We advance the `as_of` to the
-        // meet of the `upper`s of all dependencies, as we know that no replica can have produced
-        // output for that time, so we can assume that no replica `as_of` has been adanced beyond
-        // this time either.
+        // increase the wait time until the index becomes readable, and would also prevent warmup.
+        // We advance the `as_of` to `least_valid_write` - 1. (`least_valid_write` is the meet of
+        // the `upper`s of all dependencies.) This works because we know that no replica could have
+        // produced output at `least_valid_write`, so their write frontiers can't be later. Also,
+        // subtracting 1 is still ok, because we can assume that the compaction window is at least
```

Comment:
Why can you assume the compaction window is at least 1? We will soon allow users to configure the RETAIN HISTORY for everything, and I think 0 is a valid value for that.

Reply:
A while ago we discovered a bunch of things going wrong with a compaction window of 0: https://github.com/MaterializeInc/materialize/issues/13221. No idea if this looks different today, but might be worth checking if we want to allow setting it to 0 in the future?

Here we don't need to assume a non-zero compaction window for correctness, but having it at 0 means that compute reconciliation might not work.